File: C/1290-convert-binary-number-in-a-linked-list-to-integer/main.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func getDecimalValue(head *ListNode) int {
	accum := 0
	p := head
	for p != nil {
		accum = accum*2 + p.Val
		p = p.Next
	}
	return accum
}

func main() {
	fmt.Println("a")
}

File: C/1290-convert-binary-number-in-a-linked-list-to-integer/main.rs
// Definition for singly-linked list.
// #[derive(PartialEq, Eq, Clone, Debug)]
// pub struct ListNode {
//   pub val: i32,
//   pub next: Option<Box<ListNode>>
// }
//
// impl ListNode {
//   #[inline]
//   fn new(val: i32) -> Self {
//     ListNode {
//       next: None,
//       val
//     }
//   }
// }

impl Solution {
  pub fn get_decimal_value(head: Option<Box<ListNode>>) -> i32 {
    let mut h = head;
    let mut ans: i32 = 0;

    while h.is_some() {
      ans = ans * 2 + h.as_mut().unwrap().val;
      h = h.as_mut().unwrap().next.clone();
    }
    ans
  }
}
